packages/dashboard/src/components/assetModelSelection/assetModelSelect.tsx

```diff
@@ -1,44 +1,53 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 
-import { IoTSiteWiseClient } from '@aws-sdk/client-iotsitewise';
+import { IoTSiteWise } from '@aws-sdk/client-iotsitewise';
 import FormField from '@cloudscape-design/components/form-field';
 
 import { AssetForAssetModelSelect } from '../queryEditor/iotSiteWiseQueryEditor/assetModelDataStreamExplorer/assetsForAssetModelSelect/assetForAssetModelSelect';
-import { useAssetsForAssetModel } from '../queryEditor/iotSiteWiseQueryEditor/assetModelDataStreamExplorer/assetsForAssetModelSelect/useAssetsForAssetModel/useAssetsForAssetModel';
-import { useModelBasedQuery } from '../queryEditor/iotSiteWiseQueryEditor/assetModelDataStreamExplorer/modelBasedQuery/useModelBasedQuery';
+import {
+  createInitialAssetModelResource,
+  useSelectedAssetModel,
+} from '../queryEditor/iotSiteWiseQueryEditor/assetModelDataStreamExplorer/useSelectedAssetModel';
+import {
+  createInitialAssetResource,
+  useSelectedAsset,
+} from '../queryEditor/iotSiteWiseQueryEditor/assetModelDataStreamExplorer/useSelectedAsset';
+import { AssetResource } from '@iot-app-kit/react-components';
 
 type AssetModelSelectOptions = {
-  client: IoTSiteWiseClient;
+  client: IoTSiteWise;
```

Comment:
is there a reason you switched from the Client to just IoTSitewise?

Reply:
After syncing with Harry and Jason, were going to stick with the promise based client in the RE for now to avoid large refactor

Reply:
I thought the RE was already implemented with IoTSiteWise? This looks like a refactor. Was this inconsistent with the rest of the RE?

Reply:
It was initially, but we need the promise client for the internal RE components. Going to rename the prop back to 'IotSiteWiseClient' with the type of type IotSiteWise.